src/sif/pedestriancost.cc

```diff
@@ -165,29 +165,42 @@ BaseCostingOptionsConfig GetBaseCostOptsConfig() {
 
 const BaseCostingOptionsConfig kBaseCostOptsConfig = GetBaseCostOptsConfig();
 
-// Speed adjustment factors based on weighted grade. Comments here show an
-// example of speed changes based on "grade", using a base speed of 5 KPH
-// on flat roads.
-// Tobler seems a bit too "fast" uphill so we use
-// but downhill Tobler seems good
+// Cost adjustments based on variation in speed due to changes in altitude.
+// Based on the pen and paper algorithm specified in DIN 33466. For data
+// see https://gist.github.com/hungerburg/4936fa430552d625a0c8677f46492d1e
+//
+// The recipe from DIN 33466 goes: For every 300m ascent/500m descent, add
+// one hour to bucket H. For every 4km distance, add one hour to bucket D.
+// Now add half of the value in the smaller bucket to the bigger bucket
+// to get the estimated duration. The method is recommeded by the DAV and
+// the OEAV for casual hiking.
+//
+// For comparison, value researched by the Mountain Tactical Institute
 // https://mtntactical.com/research/yet-calculating-movement-uneven-terrain/
+// https://mtntactical.com/research/walking-uphill-10-grade-cuts-speed-13not-12/
+//
+// When tweaking for users with better performance, do not change flat-speed,
+// instead up the values of ascent/descent, e.g. 400/800 better matches
+// intermediate hikers. This will bring the uphill values quite close to the
+// ones from the exponential while still keeping downhill speed below flat speed.
+
 constexpr float kGradeBasedSpeedFactor[] = {
-    0.67f, // -10%  - 4.71
-    0.71f, // -8%   - 4.4
-    0.75f, // -6.5% - 4.17
-    0.8f,  // -5%   - 3.96
-    0.85f, // -3%   - 3.69
-    0.90f, // -1.5% - 3.5
-    1.0f,  // 0%    - 3.16
-    1.06f, // 1.5%  - 3.15
-    1.12f, // 3%    - 2.99
-    1.21f, // 5%    - 2.79
-    1.30f, // 6.5%  - 2.65
-    1.37f, // 8%    - 2.51
-    1.49f, // 10%   - 2.34
-    1.59f, // 11.5% - 2.22
-    1.69f, // 13%   - 2.11
-    1.82f  // 15%   - 1.97
+    1.40f,  // -10.0% - 0.67
+    1.32f,  //  -8.0% - 0.73
+    1.26f,  //  -6.5% - 0.77
+    1.20f,  //  -5.0% - 0.82
+    1.12f,  //  -3.0% - 0.89
+    1.06f,  //  -1.5% - 0.94
+    1.00f,  //   0.0% - 1.00
+    1.10f,  //   1.5% - 1.06
+    1.20f,  //   3.0% - 1.13
+    1.33f,  //   5.0% - 1.22
+    1.43f,  //   6.5% - 1.30
+    1.57f,  //   8.0% - 1.38
+    1.83f,  //  10.0% - 1.49
+    2.03f,  //  11.5% - 1.58
+    2.23f,  //  13.0% - 1.68
+    2.50f   //  15.0% - 1.82
```

Comment:
i know this is a nitpick and im sorry in advance, what is the last number in the comment supposed to mean?

i had though it was the "effective speed in MPH" after the factor is applied. so for example in the old code `-10% = 3.16 / .67 = 4.71mph` but even looking at the old values it doesnt hold true for all of them!

and in your new comment i thought maybe it was just the reciprocal of the factor so like 1/1.06  = .94 but nearer to the ends of the scales this doesnt hold.

apologies if this is obvious, for whatever reason im just not getting it at the moment!

Reply:
Sure, this is not prperly explained: The last number is the one from Mountain Tactical Institute formula, i.e. `math.exp(-0.04*slope)` ; Its written there by my script https://gist.github.com/hungerburg/4936fa430552d625a0c8677f46492d1e - it is very similar to what is there right now, and in ascent also very similar to the 400/800 variant of the DIN recipe.

Reply:
thanks for that! do you mind adding that info in comment form (the formula)?

Reply:
Don't worry, I'll do the format script for you, it's quickly done. Can you amend the comment before? Otherwise it'll likely get messed up again. Thanks!

Reply:
it was just a single space between the comment and the value, just did it with the github editor